src/ape_pm/_cli.py

```diff
@@ -0,0 +1,204 @@
+import json
+from pathlib import Path
+from typing import List, Tuple
+
+import click
+
+from ape.cli import ape_cli_context
+
+
+@click.group()
+def cli():
+    """
+    Package management tools
+    """
+
+
+@cli.command("list")
+@ape_cli_context()
+@click.option(
+    "_all", "--all", is_flag=True, help="Include packages not referenced by the local project"
+)
+def _list(cli_ctx, _all):
+    """
+    List installed packages
+    """
+
+    output_parts: List[str] = []
+
+    if _all:
+        location = cli_ctx.dependency_manager.DATA_FOLDER / "packages"
+        if location.is_dir():
+            dependencies = [x for x in location.iterdir()]
+            for dependency in dependencies:
+                output_str = click.style(dependency.name, bold=True)
+                for version_dir in dependency.iterdir():
+                    v_output_str = f"{output_str} {version_dir.name}"
+
+                    file = next(version_dir.iterdir(), None)
+                    if file:
+                        data = json.loads(file.read_text())
+                        compiled = bool(data.get("contractTypes"))
+                        if compiled:
+                            compiled_str = click.style("compiled!", fg="green")
+                            v_output_str = f"{v_output_str}, {compiled_str}"
+
+                        output_parts.append(v_output_str)
+
+    else:
+        # Limit to local project.
+        for name, versions in cli_ctx.project_manager.dependencies.items():
+            output_str = click.style(name, bold=True)
+            for version, dep in versions.items():
```

Comment:
Seems like this could be done in two steps to limit the amount of code happening in the loops, e.g. 1. collect information, 2. process information into display strings

Would make it a little easier to read

Reply:
OK I made this refactor